### ISSUE TITLE
TimeseriesChart axis reallignment

### ISSUE THREAD

**Issue by hernil** [cognitedata/gearbox.js#551]
![bilde](https://user-images.githubusercontent.com/1758308/71245745-a94e7c80-2315-11ea-89cb-ae7710bf22a6.png)

When viewing multiple timeseries in a chart it would be nice to be able to programatically reallign Y-axes. Both the center point and the scale. 

I have not looked at how Griff handles this so that would probably be guiding to how best to solve this. The most flexible would probably be to manually set center, min and max values for individual axes, however there might very well be reasons for handling more of the logic in Gearbox itself. 

I'm open for discussion! 😄

**Comment by nalaka** [comment]
@hernil - our current plan is to introduce a new prop in `TimeseriesChart` for controlling the initial Y-axes scales, say: `yAxesScales`.

This could work like how the  `hiddenSeries` prop currently works- i.e. an object with keys for each time series ID.

Something like

```typescript
yAxesScales: { [id: number]: ScaleOptions }
```
Where  I suppose `ScaleOptions` could either be an array of numbers with two elements, or an object with `min` and `max` keys.

How does that sound?

**Comment by hernil** [comment]
@nalaka This looks like a good idea as far as I can see. So by defining 
```typescript
ScaleOptions = {
    max: 100,
    min: 0
}
``` 
the center of the axis would be at 50. Correct?

**Comment by nalaka** [comment]
Exactly @hernil - in other words, y-axis would intersect x-axis at `y=0` and the top edge of y-axis will be `100`, making the mid point `50`.

**Comment by hernil** [comment]
I'm sorry to say a somewhat obvious part of this has been under-communicated on my part. 

The passing down of the props seems to work well now, but a use case for this would be syncing axes of multiple Timeseries by aligning them to one Timeserie. So we're missing a way of keeping track of the ySubdomain values for each axis. Basically the component should trigger a callback returning the new `[min, max]` array upon these values being updated by user interaction like scrolling or zooming so that the parent component can keep track of this in state. 

Assuming Griff exposes these values it should not be too hard to have Gearbox pass these on to the application 😄

**Comment by polomani** [comment]
@hernil sorry to hear that the feature didn't satisfy your needs.

We are currently not focused on new features for gearbox, and since this one is a blocker for you and also specific to your project we would suggest to add a PR on it. We'll help you to shape it and merge. That'll take much time then you'll wait for us.

**Comment by hernil** [comment]
@polomani Yes, we could do that - but it requires that Griff actually has the posibility of reporting what values are updated in the chart. Passing them further up from Gearbox should not be much of a problem if Griff supports this. Do you know if it does?

**Comment by polomani** [comment]
seems like they do have it, but in a very raw format https://griff-master.surge.sh/?path=/story/demo-linechart--on-update-domains-story

**Comment by hernil** [comment]
Yes, but that seems to be for the master branch. I couldn't find an equivalent in the 0.4 branch [except maybe](https://github.com/cognitedata/griff-react/blob/0.4/src/components/LineChart/index.js) `onZoomXaxis` but that does only half the job. In which we're back to looking at a Griff upgrade ... Or am I missing something?

**Comment by ArbenPaloja** [comment]
When will onUpdateYDomain be in gearbox.js master?

**Comment by hhenriques** [comment]
Hi @ArbenPaloja, we don't have a planned release date for it. We're planning on releasing a new version of the TimeseriesChart in beta that should address this within the next quarter.
